### PR TITLE
Pass on recipient FCM param name from extractRecipient

### LINKED ISSUE #278: InvalidRegistration when sending to topics
I'm trying to send a notification to a topic but I get the InvalidRegistration error.

This is the message's code:
```
var fcmMesssage = new gcm.Message({
		collapseKey: 'key1',
		priority: 'high',
		contentAvailable: true,
		notification: { 
			title: 'My app',
			message: 'New stuff',
			icon: "myicon",
			tag: 'mytag'
		}
	});
```

and the sender's code:
```
sender.send(fcmMesssage, { condition: '\'mytopic\' in topics'}, function (err, response) {  
		if (err){
			console.log(err);
		}
		else{
			console.log(response);
		}
	});
```

I had no problem sending the following message through the HTTP API:
```
{
    "condition":"'mytopic' in topics",
    "collapse_key":"demo",
    "priority": "high",
    "content_available":true,
    "notification":{
        "title":"My app",
        "body":"New stuff",
        "tag":"demo2"
    }
}
```

### PR DESCRIPTION
**Pull request by eladnava**
Closes #278. All tests pass.

### CODE REVIEW

**Comment by eladnava** [comment]
@hypesystem

File: lib/sender.js [hunk 1]
```diff
@@ -256,7 +248,19 @@ function extractRecipient(recipient, callback) {
...
+
+    if (key === 'condition') {
+        param = 'condition';
+    }
+    else if (['registrationIds', 'registrationTokens'].indexOf(key) !== -1) {
+        param = 'registration_ids';
+    }
+    else {
+        param = 'to';
+    }
```
Comment (hypesystem) [inline lib/sender.js]: I think I would extract this as a function:

```js
var param = getParamFromKey(key);

return nextTick(callback, null, value, param);
```

Comment (eladnava) [reply lib/sender.js]: @hypesystem Done, all tests pass.

**Comment by eladnava** [comment]
@hypesystem I agree, it could definitely be simplified, maybe by moving its logic to `getRequestBody()` in v1.

**Comment by hypesystem** [comment]
Sorry, I completely forgot this. Let's get it out :-)

**Comment by hypesystem** [comment]
Landed in 0.14.9

**Comment by eladnava** [comment]
Excellent 👍

**Pull request merged**